pkg/scaffold/templates.go

```diff
@@ -192,23 +191,23 @@ func retrieveFileTemplates(path string) (TemplateRepository, error) {
 
 // retrieveKusionTemplates retrieves the "template repository" for Kusion templates.
 // Instead of retrieving to a temporary directory, the Kusion templates are managed from
-// ~/.kusionup/current/templates.
+// ~/.kusion/templates.
 func retrieveKusionTemplates(templateName string, online bool) (TemplateRepository, error) {
 	templateName = strings.ToLower(templateName)
 
-	// Cleanup the template directory.
-	if err := cleanupLegacyTemplateDir(); err != nil {
-		return TemplateRepository{}, err
-	}
-
 	// Get the template directory.
 	templateDir, err := GetTemplateDir(ExternalTemplateDir)
 	if err != nil {
 		return TemplateRepository{}, err
 	}
 
+	// delete old external templates
+	if err = os.RemoveAll(templateDir); err != nil {
+		return TemplateRepository{}, err
+	}
+
```

Comment:
It seems that removing repos if git clone failed is much better, add a `defer` after line 218.
Besides, why using pulumi to git clone, it's such a basic function. Use a third-party package may cause unnecessary dependency problems.

Reply:
Are you intending to add `defer` with `recover` in this function? It will catch all errors and we have to distinguish which one is a git error.I moved the logic to #L215. The template directory will only be removed if the git clone fails.